src/packagedcode/windows_helpers.py

```diff
@@ -0,0 +1,311 @@
+
+        name = package_info.get('DisplayName')
+        if not name:
+            continue
+
+        version = package_info.get('DisplayVersion')
+        install_location = package_info.get('InstallLocation')
+        publisher = package_info.get('Publisher')
+
+        parties = []
+        if publisher:
+            parties.append(
+                models.Party(
+                    type=None,
+                    role='publisher',
+                    name=publisher
+                )
+            )
+
+        extra_data = {}
+        if install_location:
+            extra_data['install_location'] = install_location
+
+        yield InstalledWindowsProgram(
+            name=name,
+            version=version,
+            parties=parties,
+            extra_data=extra_data
+        )
+
+
+def reg_parse(location):
+    for installed_program in report_installed_programs(location):
+        yield installed_program
+    for installed_dotnet in report_installed_dotnet_versions(location):
+        yield installed_dotnet
+
+
+def get_installed_programs(root_dir):
+    """
+    Yield InstalledWindowsProgram objects for every detected installed program
+    from Windows registry files in known locations
+    """
+
+    # These paths are relative to a Windows docker image layer root directory
+    software_delta_loc = os.path.join(root_dir, 'Hives', 'Software_Delta')
+    SOFTWARE_reg_file_loc = os.path.join(root_dir, 'Files/Windows/System32/config/SOFTWARE')
+    software_registry_locations = [
+        software_delta_loc,
+        SOFTWARE_reg_file_loc
+    ]
+    for software_registry_loc in software_registry_locations:
+        if not os.path.exists(software_registry_loc):
+            continue
+        for installed_program in reg_parse(software_registry_loc):
+            installed_program.populate_installed_files(root_dir)
+            yield installed_program
+
+
+@attr.s()
+class InstalledWindowsProgram(models.Package):
+    filetypes = ('SOFTWARE',)
+    mimetypes = ('application/octet-stream',)
+    default_type = 'windows-program'
+
+    @classmethod
+    def recognize(cls, location):
+        for installed in reg_parse(location):
+            yield installed
+
+    def populate_installed_files(self, root_dir):
+        install_location = self.extra_data.get('install_location')
+        if not install_location:
+            return
+
+        # Manipulate install_location to fit extracted Docker image structure
+        install_location = PureWindowsPath(install_location)
+        # Remove leading drive letter ("C:\\") and replace all spaces with underscore
+        install_location_no_root = install_location.relative_to(*install_location.parts[:1])
+        install_location_no_root = str(install_location_no_root)
+        install_location_no_root = install_location_no_root.replace(' ', '_')
+        install_location_in_image = PureWindowsPath(root_dir).joinpath('Files', PureWindowsPath(install_location_no_root))
```

Comment:
I would use posix path all the way and if you get Windows path from the registry, then convert them first to POSIX the only thing we could remove is the drive letter. 
If there is some issues with inconsistencies wrt. alignment with scancode.io extracted paths, that's a bug to solve there.

Reply:
I've made the changes here: https://github.com/nexB/scancode-toolkit/blob/cab062d14f06273bc173b651e57ebb95ffa6aa32/src/packagedcode/win_reg.py#L257